controllers/ovndbcluster_controller.go

```diff
@@ -581,6 +590,76 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 			}
 		}
 	}
+
+	Log.Info("DNS Starts here")
+	if instance.Spec.DBType == v1beta1.SBDBType {
+		Log.Info("Only doing it for SB pods")
+		serviceName = ovndbcluster.ServiceNameSB
+		serviceLabels = map[string]string{
+			common.AppSelector: serviceName,
+		}
+		podList, err = ovndbcluster.OVNDBPods(ctx, instance, helper, serviceLabels)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		for _, ovnPod := range podList.Items {
+			Log.Info(fmt.Sprintf("Using Pod: %v", ovnPod.Name))
+			var dnsName string
+			var dnsIP string
+			var hostnames []string
+			dnsName = "dns-" + ovnPod.Name
+			// Get Hostname
+			svc, _ := service.GetServiceWithName(
+				ctx,
+				helper,
+				ovnPod.Name,
+				ovnPod.Namespace,
+			)
+			hostname := svc.ObjectMeta.Annotations["dnsmasq.network.openstack.org/hostname"]
+			hostnames = append(hostnames, hostname)
+
+			// Get IP
+			net_stat, _ := nad.GetNetworkStatusFromAnnotation(ovnPod.Annotations)
+			for _, v := range net_stat {
+				if v.Name == "openstack/internalapi" {
+					dnsIP = v.IPs[0]
+				}
+			}
+
+			// Create DNSRecord
+			var DNSRecords []infranetworkv1.DNSHost
+			DNSRecord := infranetworkv1.DNSHost{}
+			DNSRecord.IP = dnsIP
+			DNSRecord.Hostnames = hostnames
+			DNSRecordCname := infranetworkv1.DNSHost{}
+			DNSRecordCname.IP = dnsIP
+			DNSRecordCname.Hostnames = append(DNSRecordCname.Hostnames, "ovsdbserver-sb.openstack.svc")
```

Comment:
I wonder if the name of the DB instance CR should be part of the final hostname for DNS.

Reply:
Sorry, I don't understand what you mean

Reply:
ah, it's explained in the commit comment - so yes, EDPM nodes will use `ovsdbserver-sb.openstack.svc`, which should also resolve on the pods but use the cluster IP.

Reply:
@averdagu I mean that ""ovsdbserver-sb.openstack.svc"" should be constructed from the CR DBCluster instance name, not use a hardcoded value.

Reply:
Yeah, I have modified this, atm I have:
`headless_dns_hostname` := ovndbcluster.ServiceNameSB + "." + instance.Namespace + ".svc"`
Will wrap up some more things before adding it to this commit instead of some "wip" commit as I'm doing it now